changelog/@unreleased/pr-1568.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Implement a host metrics registry callback
```

Comment:
as a performance optimization to minimize allocation (and thereby GC) on the hot codepath of every server response